### ISSUE TITLE
Missing logs folder

### ISSUE THREAD

**Issue by syguan96** [LemonATsu/A-NeRF#3]
Hi, @LemonATsu, congratulates on your great work. 
When I tried to test the pretrained model, I met a little problem that `logs` folder cannot be found. Maybe this can be further specified in `Readme`.

**Comment by syguan96** [comment]
Also, there is another bug:
```
  File "/data/nerf/A-NeRF/core/dataset.py", line 766, in __iter__
    idx = next(sampler_iter)
  File "/data/nerf/A-NeRF/core/dataset.py", line 735, in __iter__
    yield from torch.randperm(self._n, generator=generator)
RuntimeError: Expected a 'cpu' generator device but found 'cuda:0'
```

**Comment by LemonATsu** [comment]
> 

Can you provide a specification of your environment (pytorch version, etc)? 
The default device should be cuda (as specified here: https://github.com/LemonATsu/A-NeRF/blob/main/run_nerf.py#L22). But seems like your pytorch is somehow not working on the gpu device?

**Comment by syguan96** [comment]
The return value of `torch.cuda.is_available()` is `True`. 
Here is my environment:
```
name: anerf
channels:
  - pytorch-lts
  - bottler
  - https://mirrors.sjtug.sjtu.edu.cn/anaconda/pkgs/free
  - iopath
  - fvcore
  - nvidia
  - conda-forge
  - https://mirrors.sjtug.sjtu.edu.cn/anaconda/pkgs/main
  - https://mirrors.sjtug.sjtu.edu.cn/anaconda/pkgs/main/
  - https://mirrors.sjtug.sjtu.edu.cn/anaconda/pkgs/free/
  - https://mirrors.sjtug.sjtu.edu.cn/anaconda/cloud/conda-forge/
dependencies:
  - _libgcc_mutex=0.1=main
  - _openmp_mutex=4.5=1_gnu
  - blas=1.0=mkl
  - bzip2=1.0.8=h7b6447c_0
  - ca-certificates=2021.10.26=h06a4308_2
  - certifi=2021.10.8=py37h89c1867_1
  - colorama=0.4.4=pyh9f0ad1d_0
  - cudatoolkit=11.1.74=h6bb024c_0
  - ffmpeg=4.2.2=h20bf706_0
  - freetype=2.11.0=h70c0345_0
  - fvcore=0.1.5.post20210915=py37
  - giflib=5.2.1=h7b6447c_0
  - gmp=6.2.1=h2531618_2
  - gnutls=3.6.15=he1e5248_0
  - intel-openmp=2021.4.0=h06a4308_3561
  - iopath=0.1.9=py37
  - jpeg=9b=0
  - lame=3.100=h7b6447c_0
  - lcms2=2.12=h3be6417_0
  - ld_impl_linux-64=2.35.1=h7274673_9
  - libffi=3.3=he6710b0_2
  - libgcc-ng=9.3.0=h5101ec6_17
  - libgomp=9.3.0=h5101ec6_17
  - libidn2=2.3.2=h7f8727e_0
  - libopus=1.3.1=h7b6447c_0
  - libpng=1.6.37=hbc83047_0
  - libstdcxx-ng=9.3.0=hd4cf53a_17
  - libtasn1=4.16.0=h27cfd23_0
  - libtiff=4.2.0=h85742a9_0
  - libunistring=0.9.10=h27cfd23_0
  - libuv=1.40.0=h7b6447c_0
  - libvpx=1.7.0=h439df22_0
  - libwebp=1.2.0=h89dd481_0
  - libwebp-base=1.2.0=h27cfd23_0
  - lz4-c=1.9.3=h295c915_1
  - mkl=2021.4.0=h06a4308_640
  - mkl-service=2.4.0=py37h7f8727e_0
  - mkl_fft=1.3.1=py37hd3c417c_0
  - mkl_random=1.2.2=py37h51133e4_0
  - ncurses=6.3=h7f8727e_2
  - nettle=3.7.3=hbbd107a_1
  - ninja=1.10.2=py37hd09550d_3
  - nvidiacub=1.10.0=0
  - olefile=0.46=py37_0
  - openh264=2.1.1=h4ff587b_0
  - openssl=1.1.1l=h7f8727e_0
  - pillow=8.4.0=py37h5aabda8_0
  - pip=21.2.2=py37h06a4308_0
  - portalocker=2.3.2=py37h89c1867_1
  - python=3.7.11=h12debd9_0
  - python_abi=3.7=2_cp37m
  - pytorch=1.8.2=py3.7_cuda11.1_cudnn8.0.5_0
  - readline=8.1=h27cfd23_0
  - setuptools=58.0.4=py37h06a4308_0
  - six=1.16.0=pyhd3eb1b0_0
  - sqlite=3.37.0=hc218d9a_0
  - tabulate=0.8.9=pyhd8ed1ab_0
  - tk=8.6.11=h1ccaba5_0
  - torchaudio=0.8.2=py37
  - torchvision=0.9.2=py37_cu111
  - typing_extensions=3.10.0.2=pyh06a4308_0
  - wheel=0.37.0=pyhd3eb1b0_1
  - x264=1!157.20191217=h7b6447c_0
  - xz=5.2.5=h7b6447c_0
  - yacs=0.1.6=py_0
  - yaml=0.2.5=h516909a_0
  - zlib=1.2.11=h7f8727e_4
  - zstd=1.4.9=haebb681_0
  - pip:
    - absl-py==1.0.0
    - cached-property==1.5.2
    - cachetools==4.2.4
    - charset-normalizer==2.0.9
    - chumpy==0.70
    - cloudpickle==2.0.0
    - configargparse==1.5.3
    - cycler==0.11.0
    - dask==2021.12.0
    - decorator==4.4.2
    - deepdish==0.3.7
    - fonttools==4.28.5
    - fsspec==2021.11.1
    - future==0.18.2
    - google-auth==2.3.3
    - google-auth-oauthlib==0.4.6
    - grpcio==1.43.0
    - h5py==3.6.0
    - idna==3.3
    - imageio==2.13.5
    - imageio-ffmpeg==0.4.5
    - importlib-metadata==4.10.0
    - kiwisolver==1.3.2
    - locket==0.2.1
    - markdown==3.3.6
    - matplotlib==3.5.1
    - moviepy==1.0.3
    - networkx==2.6.3
    - numexpr==2.8.1
    - numpy==1.21.5
    - oauthlib==3.1.1
    - opencv-contrib-python==4.5.5.62
    - opencv-python==4.5.5.62
    - packaging==21.3
    - partd==1.2.0
    - plotly==5.5.0
    - plyfile==0.6
    - proglog==0.1.9
    - protobuf==3.19.1
    - pyasn1==0.4.8
    - pyasn1-modules==0.2.8
    - pyglet==1.4.0b1
    - pymcubes==0.1.0
    - pyopengl==3.1.5
    - pyparsing==3.0.6
    - python-dateutil==2.8.2
    - pytorch-msssim==0.1
    - pytorch3d==0.6.1
    - pywavelets==1.2.0
    - pyyaml==5.3.1
    - requests==2.26.0
    - requests-oauthlib==1.3.0
    - rsa==4.8
    - scikit-image==0.14.2
    - scipy==1.7.3
    - smplx==0.1.28
    - tables==3.7.0
    - tenacity==8.0.1
    - tensorboard==2.7.0
    - tensorboard-data-server==0.6.1
    - tensorboard-plugin-wit==1.8.0
    - tensorboardx==1.2
    - termcolor==1.1.0
    - toolz==0.11.2
    - tqdm==4.28.1
    - trimesh==3.8.15
    - urllib3==1.26.7
    - werkzeug==2.0.2
    - zipp==3.6.0
prefix: /home/syguan/miniconda3/envs/anerf
```

**Comment by LemonATsu** [comment]
Can you try updating your pytorch to 1.9+ version? There's some bugs in older version related to `torch.randperm`.

**Comment by syguan96** [comment]
Thanks for your help! The problem was solved by upgrading PyTorch to 1.10.
